packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -120,11 +117,17 @@ export function TransferPanel() {
 
   const [transferring, setTransferring] = useState(false)
 
-  const [l1Amount, setL1AmountState] = useState<string>('')
-  const [l2Amount, setL2AmountState] = useState<string>('')
-
   const isSwitchingL2Chain = useIsSwitchingL2Chain()
 
+  // Link the amount state directly to the amount in query params -  no need of useState
+  // Both `amount` getter and setter will internally be useing useArbQueryParams functions
+  const [{ amount: queryAmount }, setQueryParams] = useArbQueryParams()
+  const amount = queryAmount && !isNaN(+queryAmount) ? queryAmount : '0'
```

Comment:
Hmm, does it makes sense to abstract this away into `useArbQueryParams`? 
That way we would always get a number returned for `amount`. If we need it as string, we can cast to it.

We could also memoize it inside `useArbQueryParams`:
```
const amount = useMemo(() => {
  if (typeof queryAmount === 'undefined' || isNaN(queryAmount)) {
    return 0
  }
  
  return parseFloat(queryAmount)
}, [queryAmount])

  const amount = queryAmount && !isNaN(+queryAmount) ? queryAmount : '0'
```

Reply:
Yes, can be done. 
However, I think we should keep the business logic after we catch the value from query params away from the `useArbQueryParams` unless it is being used extensively in a lot of different components.

Also, if we add a type-logic to `useArbQueryParams`, we might have to add some additional checks when we extend query-params for features like these - https://www.notion.so/arbitrum/Deep-link-amount-MAX-in-Bridge-UI-b583194eef5a4eeb95b274206393c9c7

I can take up your memorization suggestion within the component, coz that seems like a good optimization.

Reply:
I think it makes sense to have it in `useArbQueryParams`, because if we pull the value in two separate places we'd have to repeat the logic. 

When we add the feature you mentioned, we could set the type to be `number | 'MAX'`.

Reply:
Handled this using a custom query parser :)